hledger-lib/Hledger/Data/Types.hs

```diff
@@ -214,6 +214,7 @@ instance Eq Posting where
 -- | The position of parse errors (eg), like parsec's SourcePos but generic.
 -- File name, 1-based line number and 1-based column number.
 data GenericSourcePos = GenericSourcePos FilePath Int Int
+                      | LedgerSourcePos FilePath (Int, Int) -- ^ range of lines in ledger journal
```

Comment:
Am I understanding this commit rightly: it gives us the option to report a range of lines, rather than a single line and column position, in parse errors ?

I introduced GenericSourcePos in 2015 saying "abstract parsec's SourcePos so as to derive NFData. The NFData instance helps us time things with criterion." Now we use megaparsec, that reason may no longer be valid. Your new usage seems a reason to keep it around though, perhaps with a better name ?

The haddock needs an update.

I'd prefer to use "Journal" instead of "Ledger"; that's the term used elsewhere.